### PR TITLE
SPARQL-star built-in functions

### PR DESCRIPTION
**Pull request by hartig**
This PR extends the definition of SPARQL-star with the five built-in functions proposed in #109

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/rdf-star/pull/118.html" title="Last updated on Mar 3, 2021, 8:56 AM UTC (d3570fc)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/rdf-star/118/818755b...d3570fc.html" title="Last updated on Mar 3, 2021, 8:56 AM UTC (d3570fc)">Diff</a>

### CODE REVIEW

**Comment by hartig** [comment]
This PR is ready to be reviewed now.

@pchampin @afs @gkellogg

**Review by afs — APPROVED** [review]

**Review by pchampin — APPROVED** [review]

File: cg-spec/editors_draft.html [hunk 1]
```diff
@@ -901,6 +1327,99 @@ <h2>Translate Basic Graph Patterns</h2>
 
     </section>
 
+    <section id="builtin-functions">
+      <h2>Function Definitions</h2>
+
+      <p>SPARQL introduces operators and functions that can be used in an expression of a FILTER clause, a BIND clause, or a SELECT clause (see <a data-cite="SPARQL11-QUERY#OperatorMapping">Section&nbsp;17.3</a> and <a data-cite="SPARQL11-QUERY#SparqlOps">Section&nbsp;17.4</a> in [[SPARQL11-QUERY]]). While these operators and functions are defined to operate on <a>RDF terms</a> and <a>query variables</a>, to use them also for SPARQL-star they have to be defined to operate on <a>RDF-star terms</a> and <a>query variables</a>. To this end, when using these operators and functions in the context of SPARQL-star, their definitions as given in <a data-cite="SPARQL11-QUERY#SparqlOps">Section&nbsp;17.4</a> of [[SPARQL11-QUERY]] are extended by assuming that any mention of <code>RDF term</code> as a data type for operands is understood to be the type of all <a>RDF-star terms</a>.</p>
```
Comment (gkellogg) [inline cg-spec/editors_draft.html:1333]: SPARQL-star introduces ...?

Comment (gkellogg) [reply cg-spec/editors_draft.html:1333]: The following substitution might sound better:

`s/to use them also for SPARQL-star they have to be/for SPARQL-star they/`

File: cg-spec/editors_draft.html [hunk 2]
```diff
@@ -657,15 +657,437 @@ <h2>Grammar</h2>
...
+              <a data-cite="SPARQL11-QUERY#rExpression">Expression</a>
+              <code class="token">')'</code>
+            </div>
+            <div style="font-weight: bold">
+              `|`
+              <code class="token">'isTRIPLE'</code>
+              <code class="token">'('</code>
+              <a data-cite="SPARQL11-QUERY#rExpression">Expression</a>
+              <code class="token">')'</code>
+            </div>
```
Comment (gkellogg) [inline cg-spec/editors_draft.html:1083]: Do we need a clarifying note that, as with other SPARQL functions, they are case-insensitive? Or can this be safely assumed given the overall context?

File: cg-spec/editors_draft.html [hunk 3]
```diff
@@ -901,6 +1327,99 @@ <h2>Translate Basic Graph Patterns</h2>
...
+            <code>RDF-star term </code>
+            <code style="color:black">term1</code>,
+            <code>RDF-star term </code>
+            <code style="color:black">term2</code>,
+            <code>RDF-star term </code>
+            <code style="color:black">term3</code>
+          )
+        </p>
+
+        <p>If the 3-tuple (<code style="color:black">term1</code>, <code style="color:black">term2</code>, <code style="color:black">term3</code>) is an <a>RDF-star triple</a>, the function returns this triple. If the 3-tuple is not an RDF-star triple, then the function raises an error.</p>
```
Comment (gkellogg) [inline cg-spec/editors_draft.html:1353]: not a _valid_ RDF-star triple?

**Review by gkellogg — COMMENTED** [review]
Pretty much inline with what I had done on my own. The comments should be considered as suggestions only.

File: cg-spec/editors_draft.html (hunk 2 above)
Comment (jeenbroekstra) [reply cg-spec/editors_draft.html:1083]: There's a blanket rule in SPARQL that all keywords are case-insensitive, that automatically covers this. I have no objection to a clarifying note but I don't think it's necessary.

File: cg-spec/editors_draft.html (hunk 1 above)
Comment (jeenbroekstra) [reply cg-spec/editors_draft.html:1333]: > SPARQL-star introduces ...?

I think this first section is more generally about SPARQL, so this is fine as-is IMO. Agree with the substitution sounding better.

**Review by jeenbroekstra — COMMENTED** [review]
Mostly looks good to me, and in line with what I expected. Agree with most of @gkellogg 's editorial suggestions.

File: cg-spec/editors_draft.html (hunk 1 above)
Comment (hartig) [reply cg-spec/editors_draft.html]: > SPARQL-star introduces ...?

Not really. As also mentioned by @jeenbroekstra, this first part is indeed about SPARQL.

> The following substitution might sound better: [...]

Agreed, and fixed (commit https://github.com/w3c/rdf-star/pull/118/commits/3622b7b34769af582f61c951b5c3bc86f512914f)

File: cg-spec/editors_draft.html (hunk 2 above)
Comment (hartig) [reply cg-spec/editors_draft.html:1083]: I agree with you and @jeenbroekstra. It doesn't hurt to have such a clarifying note. Added in commit https://github.com/w3c/rdf-star/pull/118/commits/d3570fcc59f4bcb4145c945aa928edb6d1233229

File: cg-spec/editors_draft.html (hunk 3 above)
Comment (hartig) [reply cg-spec/editors_draft.html:1355]: When I wrote this sentence, I also wanted to use include the word "valid" at first. However, there is no notion validity of RDF-star triples. I think the sentence is okay as it is.  The notion of an RDF-star triple is well defined and, thus, there is no ambiguity in what this sentence means.

**Review by jeenbroekstra — APPROVED** [review]

**Comment by hartig** [comment]
As agreed during today's telco, I will merge this PR now.

I have opened a separate issue to address the concern raised by @lisp -- see #126

**Pull request merged**